grammars/ruby.cson

```diff
@@ -479,6 +479,8 @@
           |
           if|else|elsif|then|do|end|unless|while|until|or|and
         )
+        |
+        $
```

Comment:
This wouldn't be needed if we test what goes before `/` instead. This is the modified version of the begin pattern from TM's bundle:

```cson
(?:
  ^
  |
  (?<=
    [(\\[{~&=|;:,>?]|
    [\\s;]when\\s|
    [\\s;]if\\s|
    [\\s;]elsif\\s|
    [\\s;]else\\s|
    [\\s;]while\\s|
    [\\s;]unless\\s|
    [\\s;]then\\s|
    [\\s;]or\\s|
    [\\s;]and\\s|
    [\\s;]not\\s|
    [\\s.]match\\s|
    [\\s.]gsub\\s|
    [\\s.]gsub!\\s|
    [\\s.]sub\\s|
    [\\s.]sub!\\s|
    [\\s.]slice\\s|
    [\\s.]slice!\\s|
    [\\s.]index\\s|
    [\\s.]scan\\s|
    [\\s.]split\\s|
    [\\s.]partition\\s|
    [\\s.]rpartition\\s|
    [\\s;]assert_match\\s
  )
  |
  (?<=
    ^when\\s|
    ^if\\s|
    ^elsif\\s|
    ^while\\s|
    ^unless\\s
  )
)
\\s*((/))(?![*+?])
```

@50Wliu What do you think?